File: 04주차/BOJ_25497_기술연계마스터임스/BOJ_25497_기술연계마스터임스_이재만.java
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        br.readLine();
        char[] skill = br.readLine().toCharArray();
        int[] LS = {0, 0};                      // 사전기술 누적 횟수
        int answer = 0;
        for (char s : skill) {
            if ('1' <= s && s <= '9')               // 숫자이면
                answer += 1;                            // answer +1
            else if (s == 'L' || s == 'S')          // 사전기술이면
                LS[s % 2] += 1;                         // 각각의 기술 횟수 +1
            else if (LS[s % 2] == 0)                // 사전기술 없이 본 기술을 사용하는 경우
                break;                                  // 종료
            else {                                  // 본 기술 사용하는 경우
                LS[s % 2] -= 1;                         // 사전 기술 횟수 -1
                answer += 1;                            // answer +1
            }
        }
        System.out.println(answer);
    }
}

File: 04주차/BOJ_25497_기술연계마스터임스/BOJ_25497_기술연계마스터임스_이재만.py
input()
skill = input()
LS = {'L': 0, 'S': 0}
answer = 0
for s in skill:
    if s.isnumeric():
        answer += 1
    elif s in LS:
        LS[s] += 1
    elif s == 'R' and LS['L']:
        LS['L'] -= 1
        answer += 1
    elif s == 'K' and LS['S']:
        LS['S'] -= 1
        answer += 1
    else:
        break
print(answer)
